rest/oidc_api_test.go

```diff
@@ -2090,3 +2095,85 @@ func TestOpenIDConnectAuthCodeFlowWithUsernameClaim(t *testing.T) {
 		})
 	}
 }
+
+// CBG-1378 - test when OIDC provider is not reachable, and then becomes reachable
+// at a later request
+func TestEventuallyReachableOIDCClient(t *testing.T) {
+	// Modified copy of TestOpenIDConnectImplicitFlow
+	unreachableAddr := "0.0.0.0:1234"
```

Comment:
It looks like this needs to be a HTTP URL, not just an address. I think the test works but not for the reason you're expecting.

```
2021-07-20T13:48:49.532+01:00 [INF] Auth: Fetching provider config from explicitly defined discovery endpoint: <ud>0.0.0.0:1234/foo/.well-known/openid-configuration</ud>
2021-07-20T13:48:49.532+01:00 [DBG] Auth+: Fetching custom provider config from <ud>0.0.0.0:1234/foo/.well-known/openid-configuration</ud>
2021-07-20T13:48:49.532+01:00 [DBG] Auth+: Error building new request for URL <ud>0.0.0.0:1234/foo/.well-known/openid-configuration</ud>: parse "0.0.0.0:1234/foo/.well-known/openid-configuration": first path segment in URL cannot contain colon
2021-07-20T13:48:49.532+01:00 [INF] HTTP:  #001: POST /db/_session
2021-07-20T13:48:49.532+01:00 [INF] HTTP: #001:     --> 401 Invalid login  (0.2 ms)
```

Note the errors in the middle there, related to building a URL to make the OIDC config request.

Reply:
Nice spot.
Replaced address with `http://0.0.0.0` which now just refuses the connection. Could have also put a IP to time out (IE. 1.2.3.4), or a completely unreachable host (any URL that does not exist) if you think any of them seem more appropriate but ultimately leads to the same results.